### PR TITLE
[expo-camera] Convert to TypeScript

### PR DESCRIPTION
**Pull request by Szymon20000**
# Why

WIP

# How

WIP

# Test Plan

Waiting for #2626.

### CODE REVIEW

File: packages/expo-camera/src/ts-declarations/react.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import * as React from 'react';
+
+ declare module 'react' {
+   export type ElementProps<C> = C extends React.Component<infer P, any> ? P : never;
```
Comment (ide) [inline packages/expo-camera/src/ts-declarations/react.d.ts]: Use React.ComponentProps now, which replaces ElementProps: https://github.com/expo/expo/pull/2826

File: packages/expo-camera/package.json [hunk 2]
```diff
@@ -2,12 +2,25 @@
...
+  },
   "keywords": [
     "react-native",
     "expo",
     "camera"
   ],
+  "jest": {
+    "preset": "expo-module-scripts"
+  },
   "homepage": "https://docs.expo.io/",
```
Comment (sjchmiela) [inline packages/expo-camera/package.json]: Could you please also fix some `package.json` issues in this PR?
```suggestion
  "homepage": "https://github.com/expo/expo/tree/master/packages/expo-camera",
```

File: packages/expo-camera/package.json [hunk 3]
```diff
@@ -20,5 +33,8 @@
     "expo-permissions-interface": "~1.2.0",
```
Comment (sjchmiela) [inline packages/expo-camera/package.json]: Could you move `expo-*-interface` dependencies to `peerDependencies`?

File: packages/expo-camera/package.json [hunk 4]
```diff
@@ -2,12 +2,25 @@
...
+   "expo-module": "expo-module"
+  },
   "keywords": [
     "react-native",
     "expo",
     "camera"
   ],
+  "jest": {
+    "preset": "expo-module-scripts"
+  },
```
Comment (sjchmiela) [inline packages/expo-camera/package.json]: Could you also please add `repository` key?
```suggestion
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/expo/expo.git"
  },
```

**Review by sjchmiela — COMMENTED** [review]
Thanks, good job!

**Pull request merged**